sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridSerializer.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using Azure.Core;
+
+namespace Azure.Messaging.EventGrid
+{
+    internal class EventGridSerializer : IUtf8JsonSerializable
+    {
+        public object _customEvent;
+        public CancellationToken _cancellationToken;
+        public ObjectSerializer _serializer;
+
+        public EventGridSerializer(object customEvent, ObjectSerializer serializer, CancellationToken cancellationToken)
+        {
+            _customEvent = customEvent;
+            _serializer = serializer;
+            _cancellationToken = cancellationToken;
+        }
+        public void Write(Utf8JsonWriter writer)
+        {
+            var stream = new MemoryStream();
+            _serializer.Serialize(stream, _customEvent, _customEvent.GetType(), _cancellationToken);
+            stream.Seek(0, SeekOrigin.Begin);
+            JsonDocument.Parse(stream).WriteTo(writer);
+        }
```

Comment:
Might be worth filing a bug against @pakrym to find a way to skip the extra `Parse` if they're using the default `JsonObjectSerializer`.  We could possibly write a `JsonElement` into the payload more efficiently if we know it's vanilla System.Text.Json on both ends?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/13440